tokio-trace-fmt/src/default.rs

```diff
@@ -10,44 +14,163 @@ use std::fmt::{self, Write};
+///
+/// Notable default implementations of this trait are `SystemTime` and `()`. The former prints the
+/// current time as reported by `std::time::SystemTime`, and the latter does not print the current
+/// time at all. `FormatTime` is also automatically implemented for any function pointer with the
+/// appropriate signature.
+pub trait FormatTime {
+    /// Measure and write out the current time.
+    ///
+    /// When `format_time` is called, implementors should get the current time using their desired
+    /// mechanism, and write it out to the given `fmt::Write`. Implementors must insert a trailing
+    /// space themselves if they wish to separate the time from subsequent log message text.
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result;
+}
+
+impl FormatTime for () {
+    fn format_time(&self, _: &mut fmt::Write) -> fmt::Result {
+        Ok(())
+    }
+}
+
+impl FormatTime for fn(&mut fmt::Write) -> fmt::Result {
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result {
+        (*self)(w)
+    }
+}
+
+/// Retrieve and print the current wall-clock time.
+///
+/// If the `chrono` feature is enabled, the current time is printed in a human-readable format like
+/// "Jun 25 14:27:12.955". Otherwise the `Debug` implementation of `std::time::SystemTime` is used.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
+pub struct SystemTime;
+
+#[cfg(feature = "chrono")]
+impl FormatTime for SystemTime {
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result {
+        write!(w, "{} ", chrono::Local::now().format("%b %d %H:%M:%S%.3f"))
+    }
+}
+#[cfg(not(feature = "chrono"))]
+impl FormatTime for SystemTime {
+    fn format_time(&self, w: &mut fmt::Write) -> fmt::Result {
+        write!(w, "{:?} ", std::time::SystemTime::now())
+    }
+}
+
+/// Builder for a `Standard` formatter.
+#[derive(Default, Debug, Clone)]
+pub struct Builder<T = SystemTime> {
+    full: bool,
+    timer: T,
+}
+
+impl<T> Builder<T> {
+    /// Use a more verbose output format.
+    pub fn full(mut self) -> Self {
+        self.full = true;
+        self
+    }
+
+    /// Use the given `timer` for log message timestamps.
+    pub fn with_timer<T2>(self, timer: T2) -> Builder<T2>
+    where
+        T2: FormatTime,
+    {
+        Builder {
+            full: self.full,
+            timer,
+        }
+    }
+
+    /// Do not emit timestamps with log messages.
+    pub fn without_time(self) -> Builder<()> {
+        Builder {
+            full: self.full,
+            timer: (),
+        }
+    }
+
+    /// Produce a `Standard` event formatter from this `Builder`'s configuration.
+    pub fn build(self) -> Standard<T> {
+        Standard {
+            full: self.full,
+            timer: self.timer,
+        }
+    }
+}
+
+/// A pre-configured event formatter.
+///
+/// You will usually want to use this as the `Formatter` for a `FmtSubscriber`.
+#[derive(Debug, Clone)]
+pub struct Standard<T = SystemTime> {
```

Comment:
WDYT of calling this `default::Format` or something, rather than `default::Standard`? My first thought on seeing this name is "standard _what_?" 

IMO that would also more closely follow the naming elsewhere in this file --- `default::NewRecorder` is the "default implementation of `NewRecorder`, and so on).